src/antelope/stores/nfts.ts

```diff
@@ -5,241 +5,211 @@
 
 import { defineStore } from 'pinia';
 
-import { ShapedNFT } from 'src/antelope/types/NFTs';
-import { Label } from 'src/antelope/types';
+import { Label, Network, Address, IndexerTransactionsFilter, NFTClass, ERC721_TYPE } from 'src/antelope/types';
 
-import { useFeedbackStore } from 'src/antelope';
+import { useFeedbackStore, getAntelope, useAccountStore, useChainStore, useEVMStore } from 'src/antelope';
 import { createTraceFunction, isTracingAll } from 'src/antelope/stores/feedback';
+import { toRaw } from 'vue';
+import { AccountModel } from 'src/antelope/stores/account';
+import NativeChainSettings from 'src/antelope/chains/NativeChainSettings';
+import EVMChainSettings from 'src/antelope/chains/EVMChainSettings';
+import { errorToString } from 'src/antelope/config';
 
+export interface NFTsInventory {
+    owner: Address;
+    list: NFTClass[];
+    loading: boolean;
+}
+
+export interface NFTsCollection {
+    contract: Address;
+    list: NFTClass[];
+    loading: boolean;
+}
 
 export interface NftsState {
-    __nfts: {
-        [label: Label]: {
-            // each entry should be a contract address and a list of its NFTs
-            contracts: Record<string, ShapedNFT[]>,
-            accounts: Record<string, ShapedNFT[]>,
-        },
-    }
+    __filter: IndexerTransactionsFilter;
+    __inventory: Record<Label, NFTsInventory>;
+    __contracts: Record<Network, Record<Address, NFTsCollection>>;
 }
 
 const store_name = 'nfts';
 
 export const useNftsStore = defineStore(store_name, {
     state: (): NftsState => (nftsInitialState),
     getters: {
-        getAccountNfts: state => (label: Label, address: string) => state.__nfts[label].accounts[address],
+        loggedInventory: state => state.__inventory['logged']?.list,
+        loggedInventoryLoading: state => state.__inventory['logged']?.loading,
+        getInventory: state => (label: string) => state.__inventory[label],
     },
     actions: {
         trace: createTraceFunction(store_name),
         init: () => {
+            const self = useNftsStore();
             useFeedbackStore().setDebug(store_name, isTracingAll());
+            getAntelope().events.onAccountChanged.subscribe({
+                next: async ({ label, account }) => {
+                    self.__inventory[label] = {
+                        owner: account?.account || '',
+                        list: [],
+                        loading: true,
+                    };
+                    await self.updateNFTsForAccount(label, toRaw(account));
+                },
+            });
+
+            useAccountStore();
+            // update logged NFTs every 13 seconds only if the user is logged
+            setInterval(async () => {
+                if (useAccountStore().loggedAccount) {
+                    await self.updateNFTsForAccount('logged', useAccountStore().loggedAccount);
+                }
+            }, 13000);
```

Comment:
this is a little problematic because it means there is unnecessary network traffic when the user is not on an NFT page. if the user leaves an nft page and goes to wallet, we still fetch NFTs every 13 seconds. i suggest moving this interval to the component (such as NFT inventory page) rather than the store to avoid this.

Reply:
Hmmm... great observation! Let me think about alternatives. Most probably I will end up implementing what you just suggested... but let me think about it.

Reply:
@Viterbo basically i think the fetching of NFTs should be triggered by the component rather than when the user is logged in